astropy/io/votable/tree.py

```diff
@@ -3337,7 +3360,11 @@ def to_table(self, use_names_over_ids=False):
                 meta[key] = val
 
         if use_names_over_ids:
-            names = [field.name for field in self.fields]
+            names = [
+                field.name
+                for i, field in enumerate(self.fields)
+                if self._colnumbers is None or i in self._colnumbers
```

Comment:
You have defined `self._colnumbers: list[int] | None = None  # uninitialized`. Is there a reason to avoid `self._colnumbers: list[int] = []  # uninitialized`?

Reply:
Never mind, I can see that `[]` would be very different from `None`.